src/Foundation/Commerce/tests/Services/Cart/CartServiceTests.cs

```diff
@@ -0,0 +1,333 @@
+            // arrange
+            var arrange = this.ArrangeForMergeCarts(destCartSuccess: destCartSuccess);
+
+            // act
+            this.cartService.MergeCarts(this.fixture.Create<string>());
+
+            // assert
+            this.entityMapper.Received(1).Map<Result<Cart>, CartResult>(arrange.SrcCartResult);
+        }
+
+        [Fact]
+        public void MergeCarts_IfSrcCartResultIsSuccessfulAndDestCartResultIsUnsuccessful_ShouldCallMapWithDestCart()
+        {
+            // arrange
+            var arrange = this.ArrangeForMergeCarts(true);
+
+            // act
+            this.cartService.MergeCarts(this.fixture.Create<string>());
+
+            // assert
+            this.entityMapper.Received(1).Map<Result<Cart>, CartResult>(arrange.DestCartResult);
+        }
+
+        [Fact]
+        public void MergeCarts_IfSrcCartResultIsSuccessfulAndDestCartResultIsSuccessful_ShouldCallMapWithMergedCart()
+        {
+            // arrange
+            var arrange = this.ArrangeForMergeCarts(true, true);
+            // act
+            this.cartService.MergeCarts(this.fixture.Create<string>());
+
+            // assert
+            this.entityMapper.Received(1).Map<Result<Cart>, CartResult>(arrange.MergedCartResult);
+        }
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(null, "1")]
+        [InlineData("1", null)]
+        public void AddCartLine_IfParameterIsNull_ShouldThrowArgumentNullException(string productId, string variantId)
+        {
+            // act & assert
+            Assert.Throws<ArgumentNullException>(
+                () => this.cartService.AddCartLine(productId, variantId, this.fixture.Create<decimal>()));
+        }
+
+        [Fact]
+        public void AddCartLine_IfParametersAreCorrect_ShouldCallMapWithCartResult()
+        {
+            // arrange
+            var productId = this.fixture.Create<string>();
+            var variantId = this.fixture.Create<string>();
+            var quantity = this.fixture.Create<decimal>();
+            var addCartLineResult = this.fixture.Create<CartResult>();
+
+            this.cartManager.AddCartLines(
+                    Arg.Any<Sitecore.Commerce.Entities.Carts.Cart>(),
+                    Arg.Any<IEnumerable<CartLine>>())
+                .Returns(addCartLineResult);
+
+            // act
+            this.cartService.AddCartLine(productId, variantId, quantity);
+
+            // assert
+            this.entityMapper.Received(1).Map<Result<Cart>, CartResult>(addCartLineResult);
+        }
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(null, "1")]
+        [InlineData("1", null)]
+        public void UpdateCartLine_IfParameterIsNull_ShouldThrowArgumentNullException(string productId, string variantId)
+        {
+            // act & assert
+            Assert.Throws<ArgumentNullException>(
+                () => this.cartService.UpdateCartLine(productId, variantId, this.fixture.Create<decimal>()));
```

Comment:
I think we should not do `act` and `assert` at the same time.
For check exceptions we can use `[ExpectedException(typeof(ArgumentNullException))]` attribute or `Record.Exception` method to specific information from it.

Reply:
I have doubts that such approach will bring us any benefits

Reply:
https://xunit.net/docs/comparisons#note1